### PR TITLE
feat: add enableBrandProfile slackbot flag for cited audits (LLMO-5531)

### PR DESCRIPTION
**Pull request by tb-adbe**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes
- [ ] If data sources for any opportunity has been updated/added, please update the [wiki](https://wiki.corp.adobe.com/display/AEMSites/Data+Sources+for+Opportunities) for same opportunity.

## Related Issues
https://jira.corp.adobe.com/browse/LLMO-5531

Thanks for contributing!

### CODE REVIEW

File: src/utils/offsite-audit-utils.js [hunk 1]
```diff
@@ -351,7 +391,7 @@ export async function requestOffsiteScrape(context, siteId, domainScope, slackCo
       siteId,
       auditContext: {
         ...(slackContext && { slackContext }),
-        messageData: { domainScope },
+        messageData: { domainScope, ...(enableBrandProfile && { enableBrandProfile }) },
```
Comment (MysticatBot) [inline src/utils/offsite-audit-utils.js]: issue (blocking): The pattern `...(enableBrandProfile && { enableBrandProfile })` omits the field entirely when the resolved value is `false`. This is currently benign because `resolveEnableBrandProfile` defaults absent values to `false` downstream. However, it means an explicit `false` becomes indistinguishable from "never set" after the first hop - the code cannot faithfully represent an explicit opt-out.

This matters the moment the default flips to `true` (e.g. site-level config), when you need explicit `false` from Slack to override it. The same pattern appears in `scheduleDrsStatusPoll` (handler.js:783) and `triggerAnalysisAudits` (drs-status-handler.js:189).

**Fix (one-line per site):**
```js
...(enableBrandProfile != null && { enableBrandProfile })
```
This forwards both `true` and `false` while still omitting when undefined.

File: src/cited-analysis/handler.js [hunk 2]
```diff
@@ -350,7 +353,7 @@ async function runCitedAnalysisAudit(url, context, site, auditContext = {}) {
         };
       }
       log.info(`${LOG_PREFIX} No DRS content yet, requesting a scrape for top-cited`);
-      await requestOffsiteScrape(context, siteId, 'top-cited', auditContext.slackContext);
+      await requestOffsiteScrape(context, siteId, 'top-cited', auditContext.slackContext, enableBrandProfile);
```
Comment (MysticatBot) [inline src/cited-analysis/handler.js:360]: issue (blocking): Each analysis handler has two distinct code paths that call `requestOffsiteScrape` with `enableBrandProfile`:
1. `DrsNoContentAvailableError` path (tested)
2. `DrsDataNotReadyError` + `!drsScrapeRequested` path (this line - untested)

If a future refactor accidentally omits the parameter on this second path, no test would catch it. The same gap exists in the reddit and youtube handlers at the equivalent line.

**Fix:** Add a test case per handler that triggers the `DrsDataNotReadyError` path and asserts `enableBrandProfile` is forwarded on the scoped scrape request.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @tb-adbe,

**Verdict: Request changes** - two minor-effort fixes before merge.
**Complexity: HIGH** - large diff (843 lines); single-flag threading across 6 source + 6 test files.
_Changes: Threads a new `enableBrandProfile` boolean flag from Slack through the entire cited/reddit/youtube audit pipeline and the DRS scrape round-trip to reach the outgoing Mystique message (12 files)._

## Must fix before merge

1. [Important] Truthiness spread guard silently drops explicit `false` on re-enqueue - `src/utils/offsite-audit-utils.js:394` (details inline)
2. [Important] No test coverage for the second `requestOffsiteScrape` call path (DrsDataNotReadyError) - `src/cited-analysis/handler.js:356` (details inline)

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- suggestion: Consider consolidating the trailing optional parameters of `triggerAnalysisAudits` (6 positional args) and `scheduleDrsStatusPoll` (8 positional args) into an options object in a follow-up PR to prevent further accretion - `src/offsite-brand-presence/drs-status-handler.js:148`
- nit: Truncate `JSON.stringify(raw)` output in the warning log to a fixed length (e.g. 100 chars) to bound log noise from malformed payloads - `src/utils/offsite-audit-utils.js:358`
- nit: Add a test case to `resolveEnableBrandProfile` for numeric values like `0` to document whether they should resolve to `false` silently or trigger the warning - `test/utils/offsite-audit-utils.test.js`
- nit: DRS status handler test only asserts `enableBrandProfile` forwarding on `reddit-analysis`; verifying all triggered audit types carry the flag would be more robust - `test/audits/offsite-brand-presence-drs-status.test.js`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 0m 26s | <b>Cost:</b> $5.84 | <b>Commit:</b> <code>a77478a27cd2a408ffb0c95a11a8c9516197f596</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @tb-adbe,

**Verdict: Approve** - clean implementation, prior findings addressed.
**Complexity: HIGH** - large diff (12 files, flag threading across the full pipeline).
_Changes: Threads a new `enableBrandProfile` boolean flag from Slack through the cited/reddit/youtube audit pipeline and DRS scrape round-trip to reach the outgoing Mystique message (12 files)._

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- nit: Inconsistent null-guard idiom - config spreads use `!== undefined` while SQS/message spreads use `!= null`; both work identically since `resolveEnableBrandProfile` never returns `null`, but standardizing on one aids readability - `src/cited-analysis/handler.js:303`
- suggestion: Consolidate the trailing optional parameters of `scheduleDrsStatusPoll` (8 positional), `triggerAnalysisAudits` (6), and `requestOffsiteScrape` (5) into options objects in a follow-up to prevent argument-order bugs as more flags arrive - `src/offsite-brand-presence/handler.js:757`
- nit: Add a test for `{ messageData: null }` input to `resolveEnableBrandProfile` to exercise the optional-chaining path explicitly - `test/utils/offsite-audit-utils.test.js`

</details>

## Previously flagged, now resolved
- Truthiness spread guard now uses `!= null` / `!== undefined` checks consistently
- `requestOffsiteScrape` has direct unit tests covering enableBrandProfile forwarding

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 56s | <b>Cost:</b> $4.96 | <b>Commit:</b> <code>85127ce5a20bf1ee5d6ac8c9edb5c6b2b1214d88</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by AndreiAlexandruParaschiv — APPROVED** [review]
